### PR TITLE
Fixes #3625 - PHP 7.2 compatibility - functions_calendar.php

### LINKED ISSUE #3625: PHP 7 compatibility - functions_calendar.php
Can someone confirm this is an error with PHP 7.2

Warning [2] count(): Parameter must be an array or an object that implements Countable - Line: 133 - File: inc/functions_calendar.php PHP 7.2.16 (Linux)

I may be wrong, but PHP 7.2 changed the way it handles the count function when variables are empty. 

I know @count() suppresses errors to screen, but my log file is getting filled.

If this is confirmed, can !empty() be added to line 133.

Proposed:
`if((!empty($events_cache["$day-$calendar_month-$calendar_year"])) && @count($events_cache["$day-$calendar_month-$calendar_year"]) > 0)`

### PR DESCRIPTION
**Pull request by woostarUK**
Fixes #3625

### CODE REVIEW

**Comment by woostarUK** [comment]
> Why did you add brackets between the 2 expressions? You can remove them.

If I'm completely honest, I have no idea 😂🙈

Removed :)

File: inc/functions_calendar.php [hunk 1]
```diff
@@ -130,7 +130,7 @@ function build_mini_calendar($calendar, $month, $year, &$events_cache)
 
 			$link_to_day = false;
 			// Any events on this specific day?
-			if(@count($events_cache["$day-$calendar_month-$calendar_year"]) > 0)
+			if((!empty($events_cache["$day-$calendar_month-$calendar_year"])) && @count($events_cache["$day-$calendar_month-$calendar_year"]) > 0)
```
Comment (WildcardSearch) [inline inc/functions_calendar.php]: This should be:

```suggestion
			if(!empty($events_cache["$day-$calendar_month-$calendar_year"]) && @count($events_cache["$day-$calendar_month-$calendar_year"]) > 0)
```

Comment (woostarUK) [reply inc/functions_calendar.php]: I guess it's down to coding style. I was trying to keep to how empty() has been used in MyBB already.

Comment (WildcardSearch) [reply inc/functions_calendar.php]: In the line you linked to, those round brackets separate a complex condition (more than one possible passing result). In the line I am referring to in this PR, the round brackets serve no purpose and should therefore be removed. Please.

File: inc/functions_calendar.php [hunk 2]
```diff
@@ -130,7 +130,7 @@ function build_mini_calendar($calendar, $month, $year, &$events_cache)
 
 			$link_to_day = false;
 			// Any events on this specific day?
-			if(@count($events_cache["$day-$calendar_month-$calendar_year"]) > 0)
+			if(!empty($events_cache["$day-$calendar_month-$calendar_year"]) && @count($events_cache["$day-$calendar_month-$calendar_year"]) > 0)
```
Comment (dvz) [inline inc/functions_calendar.php]: Looks like the `count()` check is now redundant.

Comment (chack1172) [reply inc/functions_calendar.php]: Yes, it is redundant, we can remove it.

**Comment by woostarUK** [comment]
Oh balls. I realised I pushed this PR from my 'feature', and I've just reset my fork 🙈

**Pull request closed without merging**